### DISCUSSION TITLE
Performance issue

### DISCUSSION THREAD

**Discussion by minhducsun2002** [firefly-iii/firefly-iii#8510] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Every page load is slow.
For example, this is the loading time of the dashboard :
![image](https://github.com/firefly-iii/firefly-iii/assets/32769878/e30a9397-5f7b-4d7f-be7e-30092aa80bfc)

I have just only finished the setup, so I do not think large amount of data is the issue there. The debug page itself also takes huge amount of time to load :
![image](https://github.com/firefly-iii/firefly-iii/assets/32769878/7e6b0cf6-d419-48ca-bb48-0b90db69a61b)

### Debug information

Debug information generated at 2024-02-05 11:28:15 for Firefly III version **v6.1.7**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.1.7 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2-1+ubuntu22.04.1+deb.sury.org+1 / fpm-fcgi / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>2097152 (2 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, </td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Asia/Ho_Chi_Minh + Asia/Saigon</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-02-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-02-29 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36</td></tr></tbody></table>

### Expected behaviour

The page loads quickly, as the demo does.

### Steps to reproduce

1. Set up PHP
2. Install with `composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii 6.1.8` at `/var/www/html/`
3. Set the environment variables. I only configured MySQL credentials and nothing else.
4. Run the following :
```
php artisan migrate:refresh --seed
php artisan firefly-iii:upgrade-database
php artisan passport:install
```
5. Set up Caddy
```
my.domain.name {
        tls {
                dns cloudflare *******
        }
        encode zstd gzip
        log {
                output file /var/log/caddy/ff.log
        }
        root * /var/www/html/firefly-iii/public
        php_fastcgi unix//run/php/php8.3-fpm.sock
        file_server
}
```

### Additional info

Previously I used the Docker image version (similar configuration, only set up MySQL creds), and the problem still exists. So I believe the problem does not lie within the web server, be it Caddy or the bundled Apache in the docker image.

**Comment by JC5** [comment]
I have no clue. 12 seconds and longer is extreme. What kind of system are you running this on?

**Reply by minhducsun2002** [reply]
I'm running this on a VPS. Specs is as follows:
```
 OS: Ubuntu 22.04 jammy
 Kernel: x86_64 Linux 5.15.0-46-generic
 Uptime: 6d 2h 7m
 Packages: 671
 Shell: bash 5.1.16
 Disk: 5.2G / 903G (1%)
 CPU: Intel Xeon E5-2697 v2 @ 2.7GHz

 RAM: 671MiB / 969MiB
```
The CPU is not maxed out regardless of what I'm doing so I'm not sure. Memory isn't a problem either, it stays constant throughout requests.

Just to be sure the problem wasn't with database connection (the database is hosted in Singapore, the server hosting Firefly is in LA), I `iperf3`'d it from the machine hosting Firefly:
```
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  54.0 MBytes  45.3 Mbits/sec  11226             sender
[  5]   0.00-10.18  sec  52.3 MBytes  43.1 Mbits/sec                  receiver
```

**Reply by minhducsun2002** [reply]
I just checked a bit, and seems like Firefly calls back a lot to the database (usually this works out for other software I use), and seems like long connection times is a problem.

I spinned up a MySQL instance locally on the same machine instead and it works fine now.

Sorry for disturbance!

**Reply by JC5** [reply]
Not a problem! I've already made a note of this, lots of queries are terrible for performance indeed.
